luxonis_train/nodes/heads/ddrnet_segmentation_head.py

```diff
@@ -0,0 +1,94 @@
+"""DDRNet segmentation head.
+
+Adapted from: U{https://github.com/Deci-AI/super-gradients/blob/master/src/super_gradients/training/models/segmentation_models/ddrnet.py}
+Original source: U{https://github.com/ydhongHIT/DDRNet}
+Paper: U{https://arxiv.org/pdf/2101.06085.pdf}
+@license: U{https://github.com/Deci-AI/super-gradients/blob/master/LICENSE.md}
+"""
+
+import torch.nn as nn
+from torch import Tensor
+
+from luxonis_train.nodes.base_node import BaseNode
+from luxonis_train.nodes.blocks import ConvModule
+from luxonis_train.utils.types import LabelType
+
+
+class DDRNetSegmentationHead(BaseNode[Tensor, Tensor]):
+    in_height: int
+    in_channels: int
+    tasks: list[LabelType] = [LabelType.SEGMENTATION]
+
+    def __init__(
+        self,
+        num_classes: int,
+        in_planes: int = 128,
+        inter_planes: int = 64,
+        scale_factor: int = 8,
+        inter_mode: str = "bilinear",
+        attach_index: int = 0,
+        **kwargs,
+    ):
+        """Last stage of the segmentation network.
+
+        @type num_classes: int
+        @param num_classes: Output width.
+        @type in_planes: int
+        @param in_planes: Width of input. Defaults to 128.
+        @type inter_planes: int
+        @param inter_planes: Width of internal conv. Must be a multiple of
+            scale_factor^2 when inter_mode is pixel_shuffle. Defaults to 64.
+        @type scale_factor: int
+        @param scale_factor: Scaling factor. Defaults to 8.
+        @type inter_mode: str
+        @param inter_mode: Upsampling method. One of nearest, linear, bilinear, bicubic,
+            trilinear, area or pixel_shuffle. If pixel_shuffle is set, nn.PixelShuffle
+            is used for scaling. Defaults to "bilinear".
+        @type attach_index: int
+        @param attach_index: Index at which to attach. Defaults to 0.
+        """
+        self.attach_index = attach_index
+        super().__init__(**kwargs)
+        self.scale_factor = scale_factor
```

Comment:
Not sure if it's the same thing, but there's a helper function `luxonis_train.utils.general.infer_upscale_factor` that will compute the upscale factor based on the input shape and original image shape.
Typical usage would be:
```python
h, w = self.original_in_shape[1:]
upsacle_factor = infer_upscale_factor((self.in_height, self.in_width), (h, w))
```
where `original_in_shape`, `in_height`, and `in_widht` are again already defined `BaseNode` properties.

Reply:
@kozlov721 for this model `scale_factor` is used to scale the output to the input shape, not to the original image shape.

Reply:
Yeah that's what is meant by the original in shape. It means the input shape of the entire model. I can see how the name might be a bit confusing, implying it's the original shape of the raw image data before augmentations etc.

(just a little correction for the above snippet, the actual upscale factor would be `2 ** infer_upsalce_factor()`).

Reply:
Got, now it makes sense for me:)